src/LCT.PackageIdentifier/MavenProcessor.cs

```diff
@@ -67,6 +84,100 @@ public Bom ParsePackageFile(CommonAppSettings appSettings)
             return bom;
         }
 
+        public static void DevDependencyIdentificationLogic(List<Component> componentsForBOM, List<Component> componentsToBOM, ref List<Component> ListOfComponents)
+        {
+                       
+            List<Component> iterateBOM = componentsForBOM.Count > componentsToBOM.Count ? componentsForBOM : componentsToBOM;
+            List<Component> checkBOM = componentsForBOM.Count < componentsToBOM.Count ? componentsForBOM : componentsToBOM;
+
+      
+            ListOfComponents = DevdependencyIdentification(ListOfComponents, iterateBOM, checkBOM);
+
+        }
+
+        private static List<Component> DevdependencyIdentification(List<Component> ListOfComponents, List<Component> iterateBOM, List<Component> checkBOM)
+        {
+            foreach (var item in iterateBOM)
+            {
+                if (checkBOM.Count == 0)
+                {
+                    SetPropertiesforBOM(ref ListOfComponents, item, "false");
+                }
+                else if (checkBOM.Exists(x => x.Name == item.Name && x.Version == item.Version))
+                {
+                    SetPropertiesforBOM(ref ListOfComponents, item, "false");
+                }
+                else
+                {
+                    SetPropertiesforBOM(ref ListOfComponents, item, "true");
+
+                    BomCreator.bomKpiData.DevDependentComponents++;
+                }
+            }
+
+            return ListOfComponents;
+        }
+
+        private static void SetPropertiesforBOM(ref List<Component> componentsToBOM, Component component, string devValue)
+        {
+
+            component.Properties = new List<Property>();
+            Property isDev = new() { Name = Dataconstant.Cdx_IsDevelopment, Value = devValue };
+            Property identifierType = new() { Name = Dataconstant.Cdx_IdentifierType, Value = "Discovered" };
+            component.Properties.Add(isDev);
+            component.Properties.Add(identifierType);
+            componentsToBOM.Add(component);
+
+        }
+
+        private static void DevDependencyIdentification(List<Component> componentsForBOM, Bom bomList, ref List<Component> componentsToBOM)
+        {
+            List<Component> componentList = bomList.Components;
+
+            if (componentsForBOM?.Count >= componentList?.Count && componentsForBOM.Count != 0)
+            {
+                foreach (var entry in componentsForBOM)
+                {
+                    if (componentList.Exists(x => x.Name == entry.Name))
+                    {
+                        SetPropertiesforBOM(ref componentsToBOM, entry, "false");
+                    }
+                    else
+                    {
+                        SetPropertiesforBOM(ref componentsToBOM, entry, "true");
+
+                        BomCreator.bomKpiData.DevDependentComponents++;
```

Comment:
Check the Code Complexity for this method, maybe it's above 15

Reply:
checked it is 12